lib/spack/spack/binary_distribution.py

```diff
@@ -1644,27 +1644,15 @@ def is_backup_file(file):
                 old_prefix,
                 new_prefix,
             )
-            # Relocate links to the new install prefix
-            links = [link for link in buildinfo.get("relocate_links", [])]
-            relocate.relocate_links(links, old_layout_root, old_prefix, new_prefix)
+
+        # Relocate links to the new install prefix
+        links = [link for link in buildinfo.get("relocate_links", [])]
+        relocate.relocate_links(links, old_layout_root, old_prefix, new_prefix)
 
         # For all buildcaches
         # relocate the install prefixes in text files including dependencies
         relocate.unsafe_relocate_text(text_names, prefix_to_prefix_text)
 
-        paths_to_relocate = [old_prefix, old_layout_root]
-        paths_to_relocate.extend(prefix_to_hash.keys())
-        files_to_relocate = list(
-            filter(
-                lambda pathname: not relocate.file_is_relocatable(
-                    pathname, paths_to_relocate=paths_to_relocate
-                ),
-                map(
-                    lambda filename: os.path.join(workdir, filename),
-                    buildinfo["relocate_binaries"],
-                ),
-            )
-        )
```

Comment:
Before we were filtering `buildinfo["relocate_binaries"]`, so the semantic is different (not saying it was correct before). Why are we doing a lot of extra processing in `relocate.file_is_relocatable`? Does that processing cover some edge cases or is it plain wrong or unneeded?

Reply:
This can safely be dropped, currently it's horrible, and removing it makes Spack only better.

The old logic is as follows: take all strings minus the rpath from a binary (since `patchelf` can handle that). If among this list you find `old_prefix` or `old_layout_root`, it means this binary needs to be patched.

This is ridiculous because:

a. it's testing only *two of many* strings searched from the `prefix_to_prefix_bin` map, so it's simply incorrect;
b. it's sequential;
c. it adds a dependency on binutil's `strings` executable.

Maybe in the past this filter was useful for performance reasons, but clearly that's no longer true.